PluralKit.Bot/Commands/Member.cs

```diff
@@ -70,21 +77,115 @@ public Member(EmbedService embeds, IDatabase db, ModelRepository repo, HttpClien
+                text += $"\n{Emojis.Note} Note that this member's name contains spaces. You will need to surround it with \"double quotes\" when using commands referring to it, or just use the member's 5-character ID (which is `{member.Hid}`).";
             if (memberCount >= memberLimit)
-                await ctx.Reply($"{Emojis.Warn} You have reached the per-system member limit ({memberLimit}). You will be unable to create additional members until existing members are deleted.");
+                text += $"\n{Emojis.Warn} You have reached the per-system member limit ({memberLimit}). You will be unable to create additional members until existing members are deleted.";
             else if (memberCount >= Limits.MaxMembersWarnThreshold(memberLimit))
-                await ctx.Reply($"{Emojis.Warn} You are approaching the per-system member limit ({memberCount} / {memberLimit} members). Please review your member list for unused or duplicate members.");
+                text += $"\n{Emojis.Warn} You are approaching the per-system member limit ({memberCount} / {memberLimit} members). Please review your member list for unused or duplicate members.";
+
+            Embed? embed = null;
+            if (ctx.System.WelcomeMessage != null) {
+                var eb = new EmbedBuilder()
+                    .Title($"Welcome, {memberName}!")
+                    .Color(ctx.System.Color?.ToDiscordColor() ?? DiscordUtils.Gray)
+                    .Description(ctx.System.WelcomeMessage)
+                    .Footer(new($"System ID: {ctx.System.Hid}"));
+
+                switch (ctx.System.WelcomeMessageMode) {
+                    case WelcomeMessageMode.Off: break;
+                    case WelcomeMessageMode.Inline:
+                    {
+                        embed = eb.Build();
+                        break;
+                    }
+                    case WelcomeMessageMode.DM:
+                    {
+                        try {
+                            var dm = await ctx.Cache.GetOrCreateDmChannel(ctx.Rest, ctx.Author.Id);
+                            await ctx.Rest.CreateMessage(dm.Id, new MessageRequest{
+                                Embed = eb.Build()
+                            });
+                        }
+                        catch (ForbiddenException)
+                        {
+                            if (ctx.Channel.Type != Channel.ChannelType.Dm)
+                                text += $"\n{Emojis.Error} Could not send welcome message in DMs. Are your DMs closed?";
+                        }
+                        break;
+                    }
+                    case WelcomeMessageMode.CustomChannel:
+                    {
+                        // This shouldn't ever happen, but if it does, act as if there's no message set at all.
+                        if (ctx.System.WelcomeMessageChannel == null) break;
+
+                        try {
```

Comment:
do we really need a massive try-catch around all of this? (also, is it even necessary? I think `GetOrFetchChannel` just returns null instead of throwing an error, but I'm not sure. either way, it's better to just have a local try-catch)